plugins/main/windows/overlay/overlay_windows.go

```diff
@@ -0,0 +1,209 @@
+// Copyright 2017 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"runtime"
+	"strings"
+
+	"github.com/juju/errors"
+
+	"github.com/Microsoft/hcsshim"
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/containernetworking/cni/pkg/types/current"
+	"github.com/containernetworking/cni/pkg/version"
+	"github.com/containernetworking/plugins/pkg/hns"
+	"github.com/containernetworking/plugins/pkg/ipam"
+)
+
+type NetConf struct {
+	hns.NetConf
+
+	IPMasq            bool   `json:"ipMasq"`
+	EndpointMacPrefix string `json:"endpointMacPrefix,omitempty"`
```

Comment:
Is the endpointMacPrefix only used to ensure that the MAC address is in the administrator defined space with bit 2 set (or whatever it is)? Is there a good reason to make that prefix user-configurable?

The Linux implementation, rather than constructing the MAC address manually, sends a gratuitous ARP to notify peers of the IP/MAC mapping. Is that an option on Windows?

Reply:
Unfortunately in Windows our data plane doesn't support CA Learning i.e. inner MAC learning. So we have ARP interceptor that intercepts ARP and resolves them. so we communicate the Mapping using control plane.